plot-gui-lib/plot/main.rkt

```diff
@@ -2,12 +2,15 @@
 
 (require plot/no-gui
          "private/gui/plot2d.rkt"
-         "private/gui/plot3d.rkt")
+         "private/gui/plot3d.rkt"
+         "private/gui/snip2d.rkt")
```

Comment:
I don't think this is a good idea, because of this code in `plot-gui-lib/plot/private/gui/lazy-snip-untyped.rkt`: 

```
;; Require lazily, in case someone wants to just (require plot) in a headless setup
(lazy-require ["snip2d.rkt" (make-2d-plot-snip)]
              ["snip3d.rkt" (make-3d-plot-snip)]
              ["gui.rkt" (make-snip-frame)])
```

Adding this require in `main.rkt`, I think, defeats the point of the lazy require. (There might be some programs that `(require plot)` that will fail because they're now forced to load the gui.)

Reply:
^ but this is a guess, and I'm not sure how to test the guess

Reply:
I think you are right.  The intent is that the GUI libraries are only initialized when a plot procedure is called. Simply adding `(require plot)` to a program should not cause these to be initialized.

I'm not sure how to solve this, though.  Can you `lazy-require` a contract and a class declaration?  Or should I just provide these `for-label` to make the doc links, but not make them available to the user?

Reply:
This is why I suggested making a new module `plot/snip.rkt` that requires & provides the class & contract. Then users & the documentation (`common.rkt`) can require `plot/snip`, and programs that require `plot` don't change.

It might work to change the `require` in `plot-gui-lib/plot/main.rkt` to a lazy require.

And this might be a non-issue --- when I run an empty file `#lang racket/base (require plot)`, it seems to load the gui, so maybe the lazy require isn't helping ¯\_(ツ)_/¯